ci: Fixed an issue where GitHub Actions could not connect to PostgreSQL and MySQL.

KentarouTakeda opened the pull request:
Since the pdo-pgsql tests are only run on CircleCI when merged into the master branch, some contributors seem to be working on setting up CircleCI themselves. for example: https://github.com/php/php-src/pull/12448

As with any other test, it may be more convenient to run it in GitHub Actions.

---

(Added on 2023/10/20 0:00 JST)

### Fix details

All connections to PostgreSQL and MySQL from CI running on GitHub Actions were failing. With this fix, the following tests will now work correctly.

* pdo_pgsql
* pdo_mysql
* ~~mysqli~~
   * Hold. When fixed, tests that are no longer skipped will fail.
* pgsql

There were no problems with SQL Server, Oracle, etc., but only PostgreSQL and MySQL could not be connected. The difference is that they use GitHub Actions Service Containers.

### *Previous implementation:*

* The service name of Service Containers was set as the connection destination from the test.

If the test itself is running inside Service Containers, it can be connected without any problems by name resolution between containers. But the tests are running outside of Service Containers.

### *Implementation after fixing:*

* After exposing the connection port to the database from Service Containers to the host to `localhost`,
* Fixed to set `localhost` also for connection from test.

iluuu1994 reviewed (changes requested):
Ah, good eye. This is an oversight. However, it doesn't seem like this PR works. The number of executed tests doesn't change. Looking at the junit.out.xml, the pdo_pgsql tests are still skipped.

> PDO::__construct(): Argument #1 ($dsn) must be a valid data source name

KentarouTakeda commented:
Sorry. I had overlooked it. I'll try to fix it!

KentarouTakeda commented:
Thank you for the review. I'm sorry for taking up so much of your time.

When I corrected the point raised by @iluuu1994, I realized that there was a completely different cause than I originally thought. As a result of fixing that, many tests other than pdo_pgsql were fixed.

The concept of "fixing a test that wasn't working so that it works" remains the same, but the scope and approach of the fix has changed significantly, so we changed the subject of the pull request and added details to the text.

iluuu1994 commented on .github/workflows/push.yml:
```diff
@@ -48,19 +48,24 @@ jobs:
       MYSQL_TEST_HOST: mysql
-      PDO_MYSQL_TEST_DSN: mysql:host=mysql;dbname=test
-      PDO_MYSQL_TEST_HOST: mysql
+      PDO_MYSQL_TEST_DSN: mysql:host=localhost;dbname=test
```
If these are equivalent to the ones set in .github/actions/test-linux/action.yml there's not really a point in setting them.

KentarouTakeda replied:
Thank you, fixed it.
I also deleted `PDO_PGSQL_TEST_DSN` which was newly added to *test-linux/action* in the previous commit.

iluuu1994 reviewed (commented):
> I'm sorry for taking up so much of your time.

Please don't worry about that. Volunteer work is appreciated!

Huh. Apparently ext/mysqli, ext/pdo_mysql and ext/pdo_pgsql are all not running in GitHub actions and nobody noticed. Your PR does not yet fix the ext/mysqli tests. Do you want to have a look or should I?

KentarouTakeda commented:
> Your PR does not yet fix the ext/mysqli tests. Do you want to have a look or should I?

I tried to fix it as much as I could, but I couldn't fix everything.
Could you please take a look?

https://github.com/KentarouTakeda/php-src/actions/runs/6582187009/job/17883329034

> ========DIFF========
**##### COMMENTED This is correct on Linux**
> 001- Warning: mysqli_real_connect(): This stream does not support SSL/crypto in %s
**##### COMMENTED On Windows, this is the output**
> 001+ Warning: failed loading cafile stream: `x509.ca' in D:\a\php-src\php-src\ext\mysqli\tests\gh8978.php on line 10
>      Cannot connect to MySQL using SSL
>      done!
> ========DONE========
> FAIL Bug GH-8267 (Invalid error message when connection via SSL fails) [D:\a\php-src\php-src\ext\mysqli\tests\gh8978.phpt] 



It seems that the error message has changed in some version, but the output seems to vary depending on the OS and execution environment. I don't know how to deal with this...(Since we are not testing the OS, I feel like I should just look at `Warning`.)

Since it doesn't pass the test, I've marked it as a draft. I'm fine with either reverting the mysqli test and doing it with another pull request, or making some modifications!

KentarouTakeda commented:
If this commit is OK, I checked the all paths of the test in my repository.
https://github.com/KentarouTakeda/php-src/commit/a108e5b63b45c70cfecc38c6dbdf2a0ab145f958

iluuu1994 commented:
@KentarouTakeda Seems to have worked with a few tweaks. Thank you!

KentarouTakeda commented:
Thank you for your very kind review and merge!

I hope everyone's development experience with php-src will be improved. Let me help you again!
